src/tcgwars/logic/groovy/TcgStatics.groovy

```diff
@@ -199,6 +199,12 @@ class TcgStatics {
     bg.em().activateEffect(ef)
     return ef.selectedCards ?: []
   }
+  static CardList selectOpponentEnergy(PokemonCardSet pcs, Type...types=C) {
+    def ef = new SelectEnergy(pcs.cards, types)
+    ef.playerType = pcs.owner.opposite
```

Comment:
This should not be the default for `discardDefendingEnergyAfterDamage`. Normally the attacking player chooses the energy. If the defender is supposed to choose it this should be a completely separate method.

Reply:
With the current implementation, the defending player is choosing the energy card. Thats why i made the bugfix.

Reply:
Oh, my bad, I guess I read it backwards. I see how that is an issue now. Is there a time where the player who's turn it is is not supposed to do the selection? Might be more appropriate to change `selectEnergy` to use `bg.currentTurn` instead of `pcs.owner`.

Reply:
`discardSelfEnergyAfterDamage` uses the same method `selectEnergy`. I could imagine adding a param to `selectEnergy` but im not sure where all it is used.

Reply:
That method should have the current turn player do the discard though no? So that change should work.

Reply:
Sry I didn't misunderstood your previous comment. The `selectEnergy` method gives the selection to the owner of the pcs that it targets, rather than the player who's turn it is.

Reply:
Yeah, I think you should just change that part from `pcs.owner` to `bg.currentTurn`. No reason to make a new method, or to have a param for it.